src/tables.jl

```diff
@@ -111,26 +115,23 @@ where `names` can be a vector or tuple of String/Symbol column names, and `types
 If `temp=true`, the table will be created temporarily, which means it will be deleted when the `db` is closed.
 If `ifnotexists=true`, no error will be thrown if the table already exists.
 """
-function createtable!(db::DB, nm::AbstractString, ::Tables.Schema{names, types}; temp::Bool=false, ifnotexists::Bool=true) where {names, types}
+function createtable!(db::DB, name::AbstractString, ::Tables.Schema{names, types};
+                      temp::Bool=false, ifnotexists::Bool=true) where {names, types}
     temp = temp ? "TEMP" : ""
     ifnotexists = ifnotexists ? "IF NOT EXISTS" : ""
-    typs = [types === nothing ? "BLOB" : sqlitetype(fieldtype(types, i)) for i = 1:length(names)]
-    columns = [string(esc_id(String(names[i])), ' ', typs[i]) for i = 1:length(names)]
-    return execute(db, "CREATE $temp TABLE $ifnotexists $nm ($(join(columns, ',')))")
+    columns = [string(esc_id(String(names[i])), ' ',
+                      sqlitetype(types !== nothing ? fieldtype(types, i) : Any))
+               for i in eachindex(names)]
+    return execute(db, "CREATE $temp TABLE $ifnotexists $(esc_id(string(name))) ($(join(columns, ',')))")
 end
 
-struct TableInfo
-    exists::Bool
-    names::Vector{String}
-end
+# table info for load!():
+# returns NamedTuple with columns information,
+# or nothing if table does not exist
 function tableinfo(db::DB, name::AbstractString)
```

Comment:
I though `tableinfo()` is rather recent change, and `TableInfo` structure was not exported, so I changed it into something that looked more generally useful to me.
Before it returned `TableInfo` struct which actually contained just a vector of table column names. Now it returns the named tuple with all the information that SQLite provides, including column types etc. So the updated `tableinfo()` could be used to actually get the table information.

Also, the old version was not working, because it was checking whether the result of `columntable(...)` is a `NamedTuple` as an indication that table exists. However, `columntable()` always returns `NamedTuple` -- just in case of a missing table it has only `.q` element.